dotCMS/src/integration-test/java/com/dotmarketing/quartz/DotStatefulJobTest.java

```diff
@@ -0,0 +1,234 @@
+        }
+    }
+
+    /**
+     * Given scenario: We test that the internal methods generate properly formed names
+     * Expected Results: The call to the methods match the given patterns
+     */
+    @Test
+    public void Test_Stateful_Job_Utility_Methods(){
+        assertEquals("MyStatefulJob", getJobName(MyStatefulJob.class));
+        assertEquals("MyStatefulJob_Group", getJobGroupName(MyStatefulJob.class));
+        assertEquals("MyStatefulJob_Trigger_Group", getTriggerGroupName(MyStatefulJob.class));
+        assertTrue(DotStatefulJob.nextTriggerName(MyStatefulJob.class).startsWith("MyStatefulJob_Trigger_"));
+    }
+
+    /**
+     * create a ScheduledTask
+     * @param startDate
+     * @return
+     */
+    private ScheduledTask scheduledTask(final Date startDate){
+        final String jobName = getJobName(MyStatefulJob.class);
+        final String groupName = getJobGroupName(MyStatefulJob.class);
+        final String description = getJobDescription(MyStatefulJob.class);
+        final String nextTriggerName = DotStatefulJob.nextTriggerName(MyStatefulJob.class);
+        final String triggerGroup = getTriggerGroupName(MyStatefulJob.class);
+
+        final Map<String, Object> jobProperties = new HashMap<>();
+        //get the job detail so we dont lose any data already saved for other triggers.
+
+        final Calendar calendar = Calendar.getInstance();
+        final String cronString = new SimpleDateFormat("ss mm H d M ? yyyy")
+                .format(calendar.getTime());
+        final ScheduledTask task = new CronScheduledTask(jobName,
+                groupName, description,
+                MyStatefulJob.class.getCanonicalName(), false,
+                nextTriggerName, triggerGroup, startDate, null,
+                SimpleTrigger.MISFIRE_INSTRUCTION_FIRE_NOW, 10, true, jobProperties,
+                cronString);
+        task.setDurability(true);
+        return task;
+    }
+
+    /**
+     * Given scenario: The ScheduledTasks is build but then fired too late.
+     * Expected Results: We expect SchedulerException `Based on configured schedule, the given trigger will never fire.`
+     * @throws ParseException
+     * @throws SchedulerException
+     * @throws ClassNotFoundException
+     */
+    @Test(expected = SchedulerException.class)
+    public void Test_Schedule_Task_Make_It_Start_Later_Expect_SchedulerException()
+            throws ParseException, SchedulerException, ClassNotFoundException {
+
+        final ScheduledTask task = scheduledTask(null);
+        DateUtil.sleep(10000);
+        QuartzUtils.scheduleTask(task);
+    }
+
+    /**
+     * Given scenario: The ScheduledTasks is build but then fired too late. But this time we specify a startDate.
+     * Expected Results: The jobs fired normally.
+     * @throws ParseException
+     * @throws SchedulerException
+     * @throws ClassNotFoundException
+     */
+    @Test
+    public void Test_Schedule_Task_Make_It_Start_Later_Expect_No_Exception()
```

Comment:
Codacy found an issue: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6190817)